internal/traceprof/traceproftest/traceprof_test.go

```diff
@@ -51,20 +51,20 @@ func TestEndpointsAndCodeHotspots(t *testing.T) {
 
 			notEnoughSamples := (prof.Duration() < minCPUDuration) ||
 				(prof.LabelsDuration(CustomLabels) < minCPUDuration) ||
-				(c.CodeHotspots && prof.LabelDuration(traceprof.SpanID, "*") < minCPUDuration) ||
-				(c.AppType != Direct && c.Endpoints && prof.LabelDuration(traceprof.TraceEndpoint, "*") < minCPUDuration)
-			if attempt <= 5 && notEnoughSamples {
+				(c.CodeHotspots && prof.LabelsDuration(map[string]string{traceprof.LocalRootSpanID: res.LocalRootSpanId, traceprof.SpanID: res.SpanId}) < minCPUDuration) ||
+				(c.Endpoints && prof.LabelDuration(traceprof.TraceEndpoint, c.AppType.Endpoint()) < minCPUDuration)
+			if notEnoughSamples {
 				req.CpuDuration *= 2
 				req.SqlDuration *= 2
 				t.Logf("attempt %d: not enough cpu samples, doubling duration", attempt)
 				continue
 			}
 			require.True(t, ValidSpanID(res.SpanId))
 			require.True(t, ValidSpanID(res.LocalRootSpanId))
-			require.GreaterOrEqual(t, prof.Duration(), minCPUDuration)
-			require.GreaterOrEqual(t, prof.LabelsDuration(CustomLabels), minCPUDuration)
```

Comment:
NIT: Removing these asserts does not change the outcome of the tests, but it might reduce the quality of the error message as it's not clear which condition lead to `notEnoughSamples` to be true.

Reply:
Good point, I'll add back the asserts after the attempt loop so the failed condition can be identified.

Reply:
Looks good! Thanks!